### DISCUSSION TITLE
Accessing coefficients of polyrat.StabilizedSKRationalApproximation()

### DISCUSSION THREAD

**Discussion by kit-simmonds** [jeffrey-hokanson/polyrat#4] [Q&A] [answered]
Hi! The fit I have got from polyrat to my data looks great. 

If I'm using code of the form

`num_degree, denom_degree = 3, 3          # numerator and denominator degrees` 
`rat = polyrat.StabilizedSKRationalApproximation(num_degree, denom_degree)`
`rat.fit(x, y)`

to fit my rational function, I am assuming this fits a function of the form

`(a3*x**3 + a2*x**2 + a1*x + a0)/(b3*x**3 + b2*x**2 + b1*x + b0)`

to my data.

My question is (Hopefully this is the right place to ask this): **how do I access/return the coefficients, a3, a2, b1 e.t.c?** 

I've read in the docs here [https://polyrat.readthedocs.io/en/latest/ratfit.html](url) that `.linearized_ratfit()` returns the numerator and denominator. So far I cannot work out how to access these with `polyrat.StabilizedSKRationalApproximation()`?

Any advice would be greatly appreciated!
Thank you,
Kit

**Comment by jeffrey-hokanson** [comment, marked as answer]
The short answer is there isn't a built in way to do this because extracting the monomial coefficients is numerically fraught. (See Wilkinson's polynomial).

This isn't too much of an issue in your situation as you are using a low order approximation, but don't expect this to scale well to approximation above ~8th order.

If you do want to access these coefficients you can extract them from the numerator and denominator polynomials; e.g.., once you have `rat` as above:

```python
a = np.polyfit(X.flatten(), rat.numerator(X), num_degree)
print("a", a)
b = np.polyfit(X.flatten(), rat.denominator(X), denom_degree)
print("b", b)
```
This will yield the coefficients from lowest order to highest: i.e., a[0] + a[1] * x + a[2] * x**2 + a[3] * x**3.

**Reply by kit-simmonds** [reply]
Thank you very much for the quick reply.

Noted - fifth-degree polynomials seem to be suitable for my purposes, thanks for the warning and the example!

Kit

**Comment by gridley** [comment]
I spent so long just trying to figure this out before finding this... Thanks for asking @kit-simmonds!
